### ISSUE TITLE
Redundant $ in records

### ISSUE THREAD

**Issue by cblp** [ndmitchell/hlint#781]
Cases where ($) is redundant:

```hs
  f $ A {i = 2} -- record construction
  f $ a {i = 3} -- record update
```

**Comment by ndmitchell** [comment]
It's redundant, but deliberately not removed - turns out a lot of experienced Haskeller's find the absence of `$` there to be surprising and unwelcome so I backed off it.

**Comment by cblp** [comment]
Could it be opt-in?

**Comment by ndmitchell** [comment]
You can define it using the pattern match language - `a $ b` where `isRecordUpdate b` (or something like it) should do it. That way it can be defined outside the core of HLint.

**Comment by cblp** [comment]
yes, this custom rule works:
```yaml
- error: {lhs: a $ b, rhs: a b, side: isRecUpdate b || isRecConstr b}
```

But such behavior must be documented. I think a lot of experienced Haskeller's can have the opposite opinion.

**Comment by ndmitchell** [comment]
Yep, must be opt-in. Given that you can define such a rule yourself, I'm inclined to say that's the way you should go if you want it on your code bases.
